Ensure test coverage of remove_asmt::msc_asm_function_call

tautschnig opened the pull request:
We had coverage of the GCC counterpart to this function. This commit now adds a variant of an existing test to use MSVC assembler syntax. Running this new test with --winx64 now ensures that the code paths reaching msc_asm_function_call are hit.

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.

If your PR fixes a bug, include the regression test(s) in the same commit as the bug fix. Else, keep commits small and orthogonal, possibly placing tests in commits of their own.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

tautschnig committed:
Fix parameter handling in remove_asmt::{gcc,msc}_asm_function_call

We did not have any tests that exercised the parameter-requiring cases
of inline assembler. Enabling these uncovered that we failed to
construct the correct types. In the MSVC case we did not have any
support for parameters at all.

kroening commented:
Testing MSC assembler parsing doesn't seem to be a good fit for the set of concurrency tests.

tautschnig commented:
> Testing MSC assembler parsing doesn't seem to be a good fit for the set of concurrency tests.

Fair point, even when it does seem to be the right place to test handling of `mfence` etc. So I kept the test, but also added one to exercise `fstcw`, which in turn uncovered a few bugs (which I have now fixed).
